### DISCUSSION TITLE
file_header_template .editorconfig to take a placeholder for the file name

### DISCUSSION THREAD

**Discussion by vsfeedback** [dotnet/roslyn#47603] [General] [answered]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/idea/1135724/file-header-template-editorconfig-to-take-a-placeh.html)._

---
<p>One of my clients likes to use stylecop. One of the rules SA1638 requires all files to have a copyright header. The format it insists upon requires the header to contain the file name.</p><p>I'm trying to set up an .editorconfig file file_header_template to do this, but can't see any way to add the file name. It would be good if the template could take a placeholder where the filter name is supposed to go.</p>

---
#### Feedback Bot on 8/4/2020, 10:46 PM: 

Thank you for taking the time to provide your suggestion.&nbsp; We will do some preliminary checks to make sure we can proceed further.&nbsp; We'll provide an update once the issue has been triaged by the product team.

**Comment by sharwell** [comment, marked as answer]
Yes, this is supported. As described in [the documentation](https://docs.microsoft.com/en-us/visualstudio/ide/reference/add-file-header?view=vs-2019), the `{fileName}` placeholder may be used in the template to refer to the current file name.

**Reply by Marusyk** [reply]
what about other useful placeholders like Year, etc..

**Reply by sharwell** [reply]
Other placeholders are not supported at this time. `{currentYear}` is specifically called out in #33012 as both highly problematic and unnecessary. Discussion for this is in #44329.
